src/engine/level_script.c

```diff
@@ -360,6 +360,7 @@ static void level_cmd_clear_level(void) {
     clear_area_graph_nodes();
     clear_areas();
     main_pool_pop_state();
+    main_pool_pop_state();
```

Comment:
why we poppin twice

Reply:
```suggestion
    main_pool_pop_state();
    // the game does a push on level load and a pop on level unload, we need to add another push to store state after the level has been loaded
    main_pool_pop_state();
```
i basically just copied your message explaining why this is needed, it'd be good to explain that here so that it doesn't seem like a mistake in the future